### DISCUSSION TITLE
The "scripting" permission in version 2 of @inboxsdk/core

### DISCUSSION THREAD

**Discussion by fstn** [InboxSDK/InboxSDK#1022] [Q&A]
I migrate from 1.1.8 to 2.0.1 because of a rejection from Chrome store for remote code loading.
I had to enable chrome scripting permission to be able to load pageWorld.js into the background.js using:

`
chrome.runtime.onMessage.addListener((message, sender, sendResponse) => {
    if (message.type === 'inboxsdk__injectPageWorld' && sender.tab) {
        if (chrome.scripting) {
            chrome.scripting.executeScript({ target: { tabId: sender.tab.id }, world: 'MAIN', files: ['pageWorld.js'] });
            sendResponse(true);
            return true
        } else { // MV2 fallback.
            sendResponse(false);
        } }
});
`

Typescript code source is build with vite.js.
It's working but I cannot unable this scripting permission without loosing hundred of users... Do you have any better solution for me?

**Comment by wegry** [comment]
@fstn is your extension still using MV2 by any chance? The Chrome Web Store rejection makes it sound like MV3, but just thought I'd check.

Some background: the scripting permission was originally how we intended to differentiate between MV2 and MV3 using extensions. 

In version 1 though, in practice, if @inboxsdk/core users _didn't have_ scripting enabled, we still created a script tag in the DOM (and not through `chrome.scripting.executeScript`) even in MV3 extensions.

**Comment by fstn** [comment]
No we migrated to MV3 😥
I will try to add the script tag to pageworld.js


Thank you
